Custom Port

floaterest opened the pull request:
Passing `--static-file-host 127.0.0.1:0` as arguments to the typst preview binary will start the preview local server on a random port. This PR adds an option to the config that allows a user-specified port instead of a random one.

resolves #33

chomosuke commented:
Hello there, thanks for this PR. I'll be happy to merge this PR if you add the `port` option to README's default config also. And please explain that `0` means any free port. Please also handle the error of when the port is already taken by notifying the user with `utils.notify(...)`.

floaterest commented:
Sure, I will work on that

liuyulo commented:
Sorry for the delay.

I implemented the error handling with `utils.notify` and added documentation in README.md and man page

floaterest commented:
Sorry for the delay.

I implemented the error handling and added documention (in README.md and typst-preview.nvim.txt)

chomosuke commented on lua/typst-preview/servers/manager.lua:
```diff
@@ -32,10 +37,22 @@ function M.init(path, mode, callback)
+
+  local port = config.opts.port
+  local link = '127.0.0.1:' .. tostring(port)
+  if links[link] ~= nil then
```
I was more talking about when the port has already been used by another process. Not another typst-preview instance.

chomosuke reviewed (changes requested):
Hi, sorry for the delay as well, I did not notice that you've completed this PR.
I will be happy to merge this PR if you can somehow handle the error of if the port is already in use by another process. I believe you might need to inspect the output by the typst-preview binary to achieve this.

Thank you

floaterest commented:
Hi, I have updated the codebase.

It will now detect the "AddrInUse" error from tinymist. To handle the error, the code will attempt to use the next port (e.g. try 8001 instead of 8000).

I have also updated the docs and README.

LEXUGE commented:
Hi, can we maybe not try "port +1" by default or at least have an option to disable it? Instead can we return an error so that user can handle the error by themselves if they prefer.

chomosuke reviewed (approved):
LGTM except some early return that I subjectively did not like.

chomosuke commented:
> Hi, can we maybe not try "port +1" by default or at least have an option to disable it? Instead can we return an error so that user can handle the error by themselves if they prefer.

@LEXUGE Please open a new issue requesting this. This PR has been there for a while and i'd like to merge it instead of requesting more changes.

chomosuke commented:
And thank you for excellent work :>.
